drivers/include/periph/sdmmc.h

```diff
@@ -0,0 +1,1624 @@
+ * @return  -EBADMSG    on CRC7 error in response
+ * @return  -EINVAL     on invalid erase parameters
+ * @return  -EIO        on not further specified error incl. hardware errors
+ */
+int sdmmc_erase_blocks(sdmmc_dev_t *dev, uint32_t block_addr, uint16_t block_num);
+
+/**
+ * @brief   Read CID Register
+ *
+ * Read the CID register of a SDIO/SD/MMC Card using CMD10 and stores
+ * the results in the @p cid of type @ref sdmmc_cid_t.
+ *
+ * @pre @p dev and @p cid must not be `NULL`.
+ *
+ * @param[in]   dev     SD/MMC device to be used
+ * @param[out]  cid     CID register content
+ *
+ * @return  0           on success, @p block_num blocks were erased successfully
+ * @return  -ENODEV     if card is not present or not usable
+ * @return  -ENOTSUP    if card is in wrong state for used command
+ * @return  -EFAULT     on card status error
+ * @return  -ETIMEDOUT  on timeout condition
+ * @return  -EBADMSG    on CRC7 error in response
+ * @return  -EIO        on not further specified error incl. hardware errors
+ */
+int sdmmc_read_cid(sdmmc_dev_t *dev, sdmmc_cid_t *cid);
+
+/**
+ * @brief   Read SD Status Register
+ *
+ * Read the SD Status register of a SD Memory Card using ACMD13 and stores
+ * the results in the @p sds of type @ref sdmmc_sd_status_t.
+ *
+ * @note    This function is only available for SD Memory Cards
+ *          or the SD Memory Card portion of a combined SDIO/SD Memory Card
+ *          (Combo Card). Calling this function for a SDIO only card returns
+ *          the `-ENOTSUP` error.
+ *
+ * @pre @p dev and @p sds must not be `NULL`.
+ *
+ * @param[in]   dev     SD/MMC device to be used
+ * @param[out]  sds     SD Status register content
+ *
+ * @return  0           on success
+ * @return  -ENODEV     if card is not present or not usable
+ * @return  -ENOTSUP    if card does not support the operation
+ * @return  -EBUSY      if card is busy
+ * @return  -EFAULT     on card status error
+ * @return  -ETIMEDOUT  on timeout condition
+ * @return  -EINVAL     on invalid transfer parameters
+ * @return  -EBADMSG    on CRC7 error
+ * @return  -ENOMEM     on RX FIFO overflow error
+ * @return  -EIO        on not further specified error incl. hardware errors
+ */
+int sdmmc_read_sds(sdmmc_dev_t *dev, sdmmc_sd_status_t *sds);
+
+/**
+ * @brief   Get Capacity of SD/MMC Card
+ *
+ * Get the capacity of a SD/MMC Card device.
+ *
+ * @note    This function is only available for SD Memory Cards, MMC Cards
+ *          or the SD Memory Card portion of a combined SDIO/SD Memory Card
+ *          (Combo Card). Calling this function for a SDIO only card returns
+ *          the `-ENOTSUP` error.
+ *
+ * @pre @p dev must not be `NULL`.
+ *
+ * @param[in]   dev     SD/MMC device to be used
+ *
+ * @return  0           on success, @p block_num blocks were erased successfully
+ * @return  -ENOTSUP    if card does not support the operation
+ * @return  -EIO        on not further specified error incl. hardware errors
```

Comment:
```suggestion
 * @return              the capacity of the CD/MMC card in bytes(?) / block(?)
 * @retval  -ENOTSUP    if card does not support the operation
 * @retval  -EIO        on not further specified error incl. hardware errors
```

Note: I wasn't sure what the intended unit to report in was.

Reply:
Returns the capacity in byte or 0 on error. That's why it has to be `uint64_t` for cards > 32 GB.